tests/certbot_dns_snap_test.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+import pytest
+import subprocess
+import glob
+import os
+import re
+
+
+def _get_snap_directory():
+    snap_directory = os.environ.get('SNAP_FOLDER')
+    if not snap_directory:
+        raise ValueError('Error, SNAP_FOLDER environment variable is not set.')
+
+    return snap_directory
+
+
+def _list_dns_snaps_paths():
+    snap_directory = _get_snap_directory()
+    if not snap_directory:
+        raise ValueError('Error, SNAP_FOLDER environment variable is not set.')
+
+    return glob.glob(os.path.join(snap_directory, 'certbot-dns-*_*.snap'))
+
+
+def _extract_plugin_name(dns_snap_path):
+    return re.match(r'^.*certbot-(dns-\w+)_.*\.snap$', dns_snap_path).group(1)
```

Comment:
nit: Nothing should ever come before `certbot` here right?
```suggestion
    return re.match(r'^certbot-(dns-\w+)_.*\.snap$', dns_snap_path).group(1)
```

Reply:
In fact the paths returned by `glob` contains the relative path from which the files have been resolved, so the values are more likely to be `snap/certbot-dns-....snap` here. However we can first get the basename of the path to have a more accurate regex.

I did that.